src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubAsyncCollector.cs

```diff
@@ -112,7 +112,35 @@ public EventHubAsyncCollector(EventHubClient client)
         /// Send the batch of events.
         /// </summary>
         /// <param name="batch">the set of events to send</param>
-        protected virtual async Task SendBatchAsync(EventData[] batch)
+        protected async Task SendMultipleBatchAsync(EventData[] batch)
+        {
+            // Must sort by partition key. 
+            // All items in a call to Send() must be in the same partition. 
+            Dictionary<string, List<EventData>> buckets = new Dictionary<string, List<EventData>>();
```

Comment:
thoughts on using a ConcurrentDictionary instead of the locked _list in the `AddAsync` call to build these partition buckets, rather than after the fact? Seems like we're adding an extra step here, unless if you think this limitation may change in the future